lib/archethic/beacon_chain.ex

```diff
@@ -202,17 +165,33 @@ defmodule Archethic.BeaconChain do
   Get a beacon chain summary representation by loading from the database the transaction
   """
   @spec get_summary(binary()) :: {:ok, Summary.t()} | {:error, :not_found}
-  def get_summary(address) when is_binary(address) do
-    case TransactionChain.get_transaction(address, data: [:content]) do
-      {:ok, %Transaction{data: %TransactionData{content: content}}} ->
-        {summary, _} = Summary.deserialize(content)
+  def get_summary(summary_address) when is_binary(summary_address) do
+    case TransactionChain.get_beacon_summary(summary_address) do
+      {:ok, summary} ->
         {:ok, summary}
 
       _ ->
         {:error, :not_found}
     end
   end
 
+  @doc """
+  Get all slots of a subset from summary cache and return unique transaction summaries
+  """
+  @spec get_summary_slots(binary()) :: list(TransactionSummary.t())
+  def get_summary_slots(subset) when is_binary(subset) do
+    SummaryCache.get_current_slots(subset)
```

Comment:
If you want leverage completely streams, an idea would be to read as enumerable the ets table instead of lookup. Because lookup will load everything in memory.
For this, you can scan the ets table record and fill up the stream.
In that sense you would have a complete stream of the beacon slots